ogr/ogrsf_frmts/sqlite/ogrsqlitetablelayer.cpp

```diff
@@ -2150,7 +2164,27 @@ OGRErr OGRSQLiteTableLayer::DeleteField(int iFieldToDelete)
         eErr = m_poDS->SoftCommitTransaction();
         if (eErr == OGRERR_NONE)
         {
-            eErr = m_poFeatureDefn->DeleteFieldDefn(iFieldToDelete);
+
+            if (m_poDS->IsInTransaction())
```

Comment:
I'm not sure the IsInTransaction() logic is appropriate here. The field was unconditionally added at line 1727

Reply:
It seems correct to me: if we are in a transaction we don't want the object to be destroyed immediately (because it would crash if someone is still accessing to it) but instead we steal it and keep it alive until either the layer is destroyed or another transaction is started.  

This is was I understood (or misunderstood) to be the expected behavior.

Reply:
Disregard my comment. I misread the diff... (line 1727 is another method)